clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -299,34 +289,52 @@ private TaskHealth inferHealthNetworkBindedContainer(
     return overallTaskHealth;
   }
 
+  private TargetHealthDescription findHealthDescription(
+      List<TargetHealthDescription> targetHealths, String targetId, Integer targetPort) {
+
+    return targetHealths.stream()
+        .filter(
+            h ->
+                h.getTarget().getId().equals(targetId)
+                    && h.getTarget().getPort().equals(targetPort))
+        .findFirst()
+        .orElse(null);
+  }
+
   private TaskHealth describeTargetHealth(
-      AmazonElasticLoadBalancing amazonloadBalancing,
+      TargetHealthCacheClient targetHealthCacheClient,
       Task task,
       Service loadBalancerService,
       String serviceName,
       String targetGroupArn,
       String targetId,
       Integer targetPort,
       TaskHealth overallTaskHealth) {
-    DescribeTargetHealthResult describeTargetHealthResult =
-        amazonloadBalancing.describeTargetHealth(
-            new DescribeTargetHealthRequest()
-                .withTargetGroupArn(targetGroupArn)
-                .withTargets(new TargetDescription().withId(targetId).withPort(targetPort)));
-
-    if (describeTargetHealthResult.getTargetHealthDescriptions().isEmpty()) {
-      log.debug("Target health description is empty");
+
+    String targetHealthKey = Keys.getTargetHealthKey(accountName, region, targetGroupArn);
+    EcsTargetHealth targetHealth = targetHealthCacheClient.get(targetHealthKey);
+
+    if (targetHealth == null) {
+      log.debug("Cached EcsTargetHealth is empty for targetGroup {}", targetGroupArn);
+      evictStaleData(task, loadBalancerService);
+      return overallTaskHealth;
+    }
+    TargetHealthDescription targetHealthDescription =
+        findHealthDescription(targetHealth.getTargetHealthDescriptions(), targetId, targetPort);
+
+    if (targetHealthDescription == null) {
+      log.debug(
+          "TargetHealthDescription is empty on targetGroup '{}' for {}:{}",
+          targetGroupArn,
+          targetId,
+          targetPort);
       evictStaleData(task, loadBalancerService);
```

Comment:
I am not entirely clear on why these extra evictions happen tbh, but should we be evicting the target health key too?  Honestly no idea...